ui/models/taskcluster.js

```diff
@@ -60,9 +53,16 @@ export default class TaskclusterModel {
       const hookPayload = jsone(action.hookPayload, context);
       const { hookId, hookGroupId } = action;
       const auth = new Auth({ rootUrl: currentRepo.tc_root_url });
+
+      const userCredentials = testMode
+        ? taskcluster.getMockCredentials()
+        : taskcluster.getCredentials(currentRepo.tc_root_url);
+      if (!userCredentials) {
+        throw new Error(tcCredentialsMessage);
+      }
```

Comment:
When I try to login, I am seeing a red banner that says `Need to retrieve or renew Taskcluster credentials before action can be performed.`.

Reply:
Yes, that's deliberate to let the user know that the taskcluster action is being stopped to retrieve credentials first (so requires the user to perform the same action again once credentials have been retrieved). If the language is confusing though, I can reword it.

Although rather than stopping an action for credential retrieval, an alternative would be to do as follows:
- credential check before TC action is performed, as above
- show different text - maybe a blue 'info' banner - for when the new window opens to fetch credentials, such as 'retrieving taskcluster credentials for this repository'
- use a timeout to check for credentials again
- if credentials were added to local storage for that rootUrl, show the message and perform the action (_so it's paused rather than cancelled as per its current implementation_)
- show a variation of `There was a problem retrieving your taskcluster credentials for <rootUrl>` if credential retrieval failed

Thoughts?

Reply:
Chatted about this in IRC.  Sounds like @sarah-clements will:

1. Get TC credentials right after login
2. On action, if they're expired, catch the failure, fetch new creds and retry.

Reply:
I'll also set an expiry of 3 days for credentials.

Reply:
> Chatted about this in IRC. Sounds like @sarah-clements will:
> 
>     1. Get TC credentials right after login
To clarify, I'll fetch firefox-ci credentials after login. Servo credentials will need to be retrieved if an action is performed on that repo/rootUrl.